crates/sui-cluster-test/src/cluster.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use super::config::{ClusterTestOpt, Env};
+use async_trait::async_trait;
+use clap::*;
+use std::sync::Arc;
+use sui_config::genesis_config::GenesisConfig;
+use sui_swarm::memory::Node;
+use sui_swarm::memory::Swarm;
+use sui_types::crypto::{get_key_pair, KeyPair};
+use test_utils::network::{start_rpc_test_network, TestNetwork};
+
+const DEVNET_FAUCET_ADDR: &str = "https://faucet.devnet.sui.io:443";
+const STAGING_FAUCET_ADDR: &str = "https://faucet.staging.sui.io:443";
+const DEVNET_GATEWAY_ADDR: &str = "https://gateway.devnet.sui.io:443";
+const STAGING_GATEWAY_ADDR: &str = "https://gateway.staging.sui.io:443";
+const DEVNET_FULLNODE_ADDR: &str = "https://fullnode.devnet.sui.io:443";
+const STAGING_FULLNODE_ADDR: &str = "https://fullnode.staging.sui.io:443";
+
+pub struct ClusterFactory;
+
+impl ClusterFactory {
+    pub async fn start(
+        options: &ClusterTestOpt,
+    ) -> Result<Arc<dyn Cluster + Sync + Send>, anyhow::Error> {
```

Comment:
you probably don't want this to be an Arc but instead a Box, because otherwise you can't have any methods that require &mut self.